lib/commands/interactive.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Start interactive REPL
+ */
+
+const path = require('path');
+const repl = require('../testLauncher/repl');
+const logger = require('../utils/logger');
+const {getShortFunctionBody} = require('../utils/stringUtils');
+const texts = require('../texts');
+const {stackTraceParser} = require('../utils/stackTraceParser');
+const {authContext} = require('../context');
+
+let isReplActive = false;
+
+/**
+ * Prepares parameters for the REPL session
+ * @param {{ignored: *, watch: *, repeater: Function? cwd: String?, vars: Array?}} opts
+ * @returns {*[]}
+ */
+function getReplParameters(opts) {
+	// get dir where .interactive() was called
+	const callFile = path.normalize(stackTraceParser(new Error())[2].file);
+	const cwd = path.dirname(callFile);
+	const files = opts.watch || path.join(cwd, '**/*.js');
+	const watch = Array.isArray(files)? files : [files? files : cwd];
+	const suitest = this;
+	const vars = Object.assign({suitest}, opts.vars);
+
+	return [vars, cwd, watch];
+}
+
+/**
+ * Start interactive REPL
+ * @param {{ignored: *, watch: *, repeater: Function? cwd: String?, vars: Array?}} opts
+ * @returns {Promise} response object
+ */
+async function interactive(opts = {}) {
+	if (isReplActive)
+		return Promise.resolve();
+
+	const {replWarnInteractive, replWelcomeMessage, replSessionEnded} = texts;
+
+	if (!authContext.isInteractiveSession()) {
+		logger.warn(replWarnInteractive());
+
+		return Promise.resolve();
```

Comment:
No need to return Promise.resolve() because function already async and all returned values will be wrapped in Promise.